test/common.py

```diff
@@ -218,21 +220,26 @@ def assertTensorsEqual(a, b):
             else:
                 assertTensorsEqual(x, y)
         elif isinstance(x, string_classes) and isinstance(y, string_classes):
-            super(TestCase, self).assertEqual(x, y)
+            super(TestCase, self).assertEqual(x, y, message)
         elif type(x) == set and type(y) == set:
-            super(TestCase, self).assertEqual(x, y)
+            super(TestCase, self).assertEqual(x, y, message)
         elif is_iterable(x) and is_iterable(y):
-            super(TestCase, self).assertEqual(len(x), len(y))
+            super(TestCase, self).assertEqual(len(x), len(y), message)
             for x_, y_ in zip(x, y):
                 self.assertEqual(x_, y_, prec, message)
-        else:
-            try:
-                self.assertLessEqual(abs(x - y), prec, message)
+        elif isinstance(x, Number) and isinstance(y, Number):
+            if abs(x) == float('inf') and abs(y) == float('inf'):
```

Comment:
I’m wondering if we really want to allow comparing infs. They can represent arbitrary overflows and if you write your test incorrectly such that it always overflows then it will pass and you won’t even notice that it doesn’t check anything.

Reply:
I removed the `inf` checks, but we are doing the approximate test now only if `x != y` which indirectly does account for infinities, i.e. ( `float('inf') == float('inf')  => True` in python ). There is a test in `test_autograd` which needs this and I think it may be useful in comparing `log_prob` values when probability is 0. 

> if you write your test incorrectly such that it always overflows then it will pass and you won’t even notice that it doesn’t check anything

This could happen if the expected value is not set by the user (to infinity) but rather computed through some other method. Let me know what you think.